### PR TITLE
jobs: use 0xC player stats log line to determine gcd 

### PR DESCRIPTION
**Pull request by KattTails**
Hiya,

I've added 0xC log line support for the jobs overlay as requested in #595. It runs the setupJob() functions with a default 2.5s gcd, then as soon as we see `0C: Player stats:` (usually instantly, unless you reload the overlay manually) it updates the relevant variables. It has support for every status affecting GCDs; I added bare bones support for BRD, NIN, WHM and SAM purely for feature completion.

I'm not really confident that the statChangeRegex in `regexes.js` matches the surrounding functions' style, and I'm open to feedback on that especially.

### CODE REVIEW

File: resources/lang/de.js [hunk 1]
```diff
@@ -36,13 +36,19 @@ class CactbotLanguageDe extends CactbotLanguage {
       Devotion: 'Hingabe', // 0x4bd
       FoeRequiem: 'Requiem Der Feinde', // up 0x8b, down 0x8c
       LeadenFist: 'Verbesserter Schlag auf Schlag',
+      StormsEye: 'Sturmbrecher',
```
Comment (KattTails) [inline resources/lang/de.js]: I added a couple of missing translations too.

File: ui/jobs/jobs.js [hunk 2]
```diff
@@ -293,8 +360,6 @@ function setupRegexes() {
   ]);
 }
 
-let kMeleeWithMpJobs = ['DRK', 'PLD'];
```
Comment (KattTails) [inline ui/jobs/jobs.js]: I moved this up as it felt out of place being buried so far down in the code.

File: ui/jobs/jobs.js [hunk 3]
```diff
@@ -1283,7 +1371,7 @@ class Bars {
   }
 
   setupPld() {
-    let gcd = this.options.PldGcd;
+    let gcd = 2.5;
```
[the remark this answers is not in the document]
Comment (KattTails) [reply ui/jobs/jobs.js]: done

File: ui/jobs/jobs.js [hunk 4]
```diff
@@ -75,6 +71,77 @@ let kYouLoseEffectRegex = null;
...
+  [302, 302], [311, 311], [322, 322], [331, 331], [341, 341],
+  [342, 393], [344, 444], [345, 496], [346, 548], [347, 600],
+  [349, 651], [350, 703], [351, 755], [352, 806], [354, 858],
+  [355, 941], [356, 1032], [357, 1133], [358, 1243], [369, 1364],
+  [360, 1497], [361, 1643], [362, 1802], [363, 1978], [364, 2170],
+  [365, 2263], [366, 2360], [367, 2461], [368, 2566], [370, 2676],
+  [372, 2790], [374, 2910], [376, 3034], [378, 3164], [380, 3300]];
+
+// Source: http://theoryjerks.akhmorning.com/guide/speed/
+function kCalcGCDFromStat(stat, actiondelay) {
```
[the remark this answers is not in the document]
Comment (KattTails) [reply ui/jobs/jobs.js]: changed as requested.

File: resources/regexes.js [hunk 5]
```diff
@@ -319,6 +319,27 @@ var Regexes = {
     return Regexes.parse(str);
   },
 
+  // matches: https://github.com/quisquous/cactbot/blob/master/docs/LogGuide.md#0c-playerstats
+  statChange: () => {
+    let str = '\\y{Timestamp} 0C:Player Stats: ' + '\\d+' + ':' + // skip job field
+      Regexes.namedCapture('strength', '\\d+') + ':' +
```
[the remark this answers is not in the document]
Comment (KattTails) [reply resources/regexes.js]: I've had a go at matching it with the other functions and adding a test.

File: resources/regexes.js [hunk 6]
```diff
@@ -319,6 +319,53 @@ var Regexes = {
...
+      f = {};
+    validateParams(f, 'statChange', [
+      'job',
+      'strength',
+      'dexterity',
+      'vitality',
+      'intelligence',
+      'mind',
+      'piety',
+      'attack_power',
```
Comment (quisquous) [inline resources/regexes.js]: sorry, one more naming nit: `attackPower` etc instead of `attack_power`

File: ui/jobs/jobs.js [hunk 7]
```diff
@@ -1726,11 +1789,134 @@ class Bars {
...
+        break;
+      case 4:
+        type2Buffs += 12;
+        break;
+      }
+    }
+    // TODO: this probably isn't useful to track
+    let astralUmbralMod = 100;
+
+    let GCDms = Math.floor(1000 - Math.floor(130 * (stat - kLevelMod[this.level][0]) /
```
Comment (KattTails) [inline ui/jobs/jobs.js]: It felt neater that you could just pass `this.level` rather than subtracting one so I bumped the `kLevelMod` entries to match.

**Comment by KattTails** [comment]
I've renamed as suggested, it should be ready to go now.

File: ui/jobs/jobs.js (hunk 7 above)
Comment (quisquous) [reply ui/jobs/jobs.js]: :+1:

**Comment by quisquous** [comment]
Thanks!!

**Pull request merged**